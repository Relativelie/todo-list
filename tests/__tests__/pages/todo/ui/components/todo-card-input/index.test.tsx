import React from "react";
import TodoComponentsTodoCardInput from "../../../../../../../src/pages/todo/ui/screens/main";
import { render } from "../../../../../../testsSetup/test-utils";
import { fireEvent, screen } from "@testing-library/dom";
import userEvent from "@testing-library/user-event";

let addListFunc: Function;
beforeEach(() => {
  addListFunc = jest.fn();
});

describe("Components Todo Card Input", () => {
    test("snapshot - Components Todo Card Input", () => {
      const tree = render(<TodoComponentsTodoCardInput />);
      expect(tree).toMatchSnapshot();
    });

    test("Components Todo Card Input - click to input", () => {
      const { getByTestId } = render(<TodoComponentsTodoCardInput />);
      const inputEl = getByTestId("task-input-click") as HTMLInputElement;
      userEvent.click(inputEl);
      expect(inputEl.value).toBe(undefined);

      expect(inputEl.className).toBe(
        "MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-adornedStart css-1vdbjvs-MuiInputBase-root-MuiInput-root"
      );
    });

  test("Components Todo Card Input - input value", () => {
    const { getByTestId } = render(<TodoComponentsTodoCardInput />);
    const contentInput = getByTestId("task-input");
    fireEvent.change(contentInput, {
      target: { value: "some text for input 123!@#$%^&*()_+" },
    });
    const inputEl = getByTestId("task-input") as HTMLInputElement;
    expect(inputEl.value).toBe("some text for input 123!@#$%^&*()_+");
  });

  test("Components Todo Card Input - enter non-empty value", () => {
    const { getByTestId } = render(<TodoComponentsTodoCardInput />);
    const contentInput = getByTestId("task-input");
    fireEvent.change(contentInput, {
      target: { value: "some text for input 123!@#$%^&*()_+" },
    });
    const inputEl = getByTestId("task-input") as HTMLInputElement;
    fireEvent.keyPress(inputEl, { key: 'Enter', keyCode: 13 });
    expect(inputEl.value).toBe("");
  });
});
