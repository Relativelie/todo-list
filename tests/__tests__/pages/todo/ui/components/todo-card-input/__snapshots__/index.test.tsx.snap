// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Todo Card Input snapshot - Components Todo Card Input 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1oskuwq"
      >
        <div
          class="MuiTypography-root MuiTypography-h1 css-1fmrqoi-MuiTypography-root"
        >
          todos
        </div>
        <div
          class="css-1a2x4g5"
        >
          <div
            class="MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedStart css-1vdbjvs-MuiInputBase-root-MuiInput-root"
            data-testid="task-input-click"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreSharpIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41z"
                />
              </svg>
            </div>
            <input
              class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1x51dt5-MuiInputBase-input-MuiInput-input"
              data-testid="task-input"
              placeholder="What needs to be done?"
              type="text"
              value=""
            />
          </div>
          <div
            class="css-ur7kwn"
          >
            <div
              class="css-1q2lbmy"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-18j8e14-MuiTypography-root"
              >
                List is empty
              </p>
            </div>
          </div>
          <div
            class="css-1ursc7l"
          >
            <div
              class="css-17vaav6"
            />
            <div
              class="css-8ksf2s"
            >
              <div
                class="css-20wnlw"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-qab2vh-MuiTypography-root"
                >
                  All
                </span>
              </div>
              <div
                class="css-1rexy4l"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-qab2vh-MuiTypography-root"
                >
                  Active
                </span>
              </div>
              <div
                class="css-1rexy4l"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-qab2vh-MuiTypography-root"
                >
                  Completed
                </span>
              </div>
            </div>
            <div
              class="css-rf9zus"
            >
              <div
                class="css-1rexy4l"
              >
                <span
                  class="MuiTypography-root MuiTypography-caption css-qab2vh-MuiTypography-root"
                >
                  Clear completed
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
       
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1oskuwq"
    >
      <div
        class="MuiTypography-root MuiTypography-h1 css-1fmrqoi-MuiTypography-root"
      >
        todos
      </div>
      <div
        class="css-1a2x4g5"
      >
        <div
          class="MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedStart css-1vdbjvs-MuiInputBase-root-MuiInput-root"
          data-testid="task-input-click"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreSharpIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41z"
              />
            </svg>
          </div>
          <input
            class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1x51dt5-MuiInputBase-input-MuiInput-input"
            data-testid="task-input"
            placeholder="What needs to be done?"
            type="text"
            value=""
          />
        </div>
        <div
          class="css-ur7kwn"
        >
          <div
            class="css-1q2lbmy"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-18j8e14-MuiTypography-root"
            >
              List is empty
            </p>
          </div>
        </div>
        <div
          class="css-1ursc7l"
        >
          <div
            class="css-17vaav6"
          />
          <div
            class="css-8ksf2s"
          >
            <div
              class="css-20wnlw"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-qab2vh-MuiTypography-root"
              >
                All
              </span>
            </div>
            <div
              class="css-1rexy4l"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-qab2vh-MuiTypography-root"
              >
                Active
              </span>
            </div>
            <div
              class="css-1rexy4l"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-qab2vh-MuiTypography-root"
              >
                Completed
              </span>
            </div>
          </div>
          <div
            class="css-rf9zus"
          >
            <div
              class="css-1rexy4l"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-qab2vh-MuiTypography-root"
              >
                Clear completed
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
     
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
